src/MySqlConnector/MySqlClient/ConnectionPool.cs

```diff
@@ -53,11 +53,26 @@ internal sealed class ConnectionPool
 			}
 		}
 
+		private bool SessionIsHealthy(MySqlSession session)
+		{
+			if (!session.IsConnected)
+				return false;
+			if (session.PoolGeneration != m_generation)
+				return false;
+			if (session.DatabaseOverride != null)
+				return false;
+			if (m_connectionSettings.ConnectionLifeTime > 0
+			    && DateTime.UtcNow - session.CreatedUtc > TimeSpan.FromSeconds(m_connectionSettings.ConnectionLifeTime))
```

Comment:
Nitpick: I might personally write `(DateTime.UtcNow - session.CreatedUtc).TotalSeconds >= m_connectionSettings.ConnectionLifeTime` (slightly fewer keystrokes, avoids the creation of a temporary) but it's hardly worth changing.